### DISCUSSION TITLE
Core1262-868M connection advice

### DISCUSSION THREAD

**Discussion by devRomeos** [jgromes/RadioLib#1265] [Q&A]
Hello, I have two waveshare modules and I'm struggling to get them to work at medium range. I think the key is the TX/RXEN that I'm not sure I'm doing that right. from [waveshare](https://www.waveshare.com/wiki/Core1262-868M) I see:

> RXEN, TXEN are RF single-pole switch pins
> RXEN low level, TXE high level, SX1262 is in receiving mode. 
> RXEN high level, TXEN low level, SX1262 is in transmitting mode.

However I am seeing just the opposite, so just picking receiving  as example, I observe more signal when pull RXEN to VCC  and TXE to GND. Can anyone confirm the real Low/High combination for both modes.

> DIO2 is connected to RXEN Set as the control pin of the RF single-pole switch

From what I understand, DIO2 has its own task internally so I have not connected it to my controller. is this correct?

By otherhand, can any one suggest me what RSSI I should get at 1 meter? I get arround -30dB (10dbm output & 2dBi ants) when both modules are each other, however as soon I move one of them 1 metre away I get -50dB or more that seems too low. I am new to this 868 sub-band so don't know exactly what to expect.

Finally, I have lost some confidence in this brand/module in these days struggling, if anyone knows any SX1262 board like this, with **integrated uFL or SMA connector and with a RX consumption of ~6mA** please let me know.

Thanks

**Comment by jgromes** [comment]
The naming seems to suggest the info on Waveshare website is indeed incorrect - `RXEN` typically stands for RX ENable. Same with TXEN. The mention of DIO2 being connected to the RF switch is also very confusing. DIO2 can indeed be used to control an RF switch automatically, this feature is enabled in RadioLib by default. However, in that setup, there is usually no need to have any external control pins. And from the schematic on waveshare website (https://files.waveshare.com/upload/c/c1/CoreSX1262_Sch.pdf), DIO2 seems unconnceted and left to the user.

**Reply by devRomeos** [reply]
Thank you very much for your support, I guess a company that doesn't even know how to explain its own product is not a sign of quality.

I will wait for someone who has used it to confirm these bad results.

**Comment by HeadBoffin** [comment]
Using many of the well marketed, well intentioned Far East offerings usually comes with the penalty of fiddling with them until they work, usually based on lots of prior experience of the subject matter, schematic reading, test probes and a few hundred litres of midnight oil. But when you get it working, they are often good value - setup & making the modules is too expensive even for them to mess about with a poor design. But that's no good if we are burning through battery needlessly:

> Finally, I have lost some confidence in this brand/module in these days struggling, if anyone knows any SX1262 board like this, 
> with integrated uFL or SMA connector and with a RX consumption of ~6mA please let me know.

I've searched high & low for such an item and am yet to have an a-ha moment. I can find boards with integrated MCU's - so if a STM32WL or a nRF52840 or an EFR32 attached is not an issue, and in the grand scheme of things there are bigger challenges in device creation than using a different IDE but with the same code base (mostly LoRa Basics Modem), or if you are lucky, comes with an Arduino port so you can use RadioLib directly or if you are even luckier, has RadioLib ported to it's native environment.

In the meanwhile, I've become sufficiently desperate for a SX1261 module, the EU low power variant, that I can wire to whatever I want/need/like, that I have started the first steps towards making my own. This is entirely feasible with larger passives (for my manufacturing capabilities) and it doesn't need an RF switch so is likely to be slightly larger than the "standard" sizes we see, but not much bigger.

The alternative is to buy the SX1261MB2BAS dev board - huge but fundamentally just the radio module - it's the Semtech reference design which I will be copying as close as the slight larger passives will allow.
